src/hotspot/share/opto/parseHelper.cpp

```diff
@@ -358,6 +358,15 @@ void PEAState::add_new_allocation(Node* obj) {
 
   if (nfields >= 0) {
     AllocateNode* alloc = obj->in(1)->in(0)->as_Allocate();
+#ifndef PRODUCT
+    if (PEA_debug_idx != 0) {
```

Comment:
looks like this if statement is not required. Checks covered by nested if-else.

Reply:
yes. it's left-over code.